cmake/Modules/FindONEMKL.cmake

```diff
@@ -0,0 +1,54 @@
+set(ONEMKL_XPU_FOUND FALSE)
+
+set(ONEMKL_XPU_LIBRARIES)
+
+if(DEFINED ENV{MKLROOT})
+  set(ONEMKL_ROOT $ENV{MKLROOT})
+endif()
+
+if(NOT ONEMKL_ROOT)
+  message(WARNING "Cannot find oneMKL in ENV{MKLROOT}, please setup oneMKL before building!!")
+  return()
+endif()
+
+find_file(
+  ONEMKL_INCLUDE_DIR
+  NAMES include
+  HINTS ${ONEMKL_ROOT}
+  NO_DEFAULT_PATH
+)
+
+find_file(
+  ONEMKL_LIB_DIR
+  NAMES lib
+  HINTS ${ONEMKL_ROOT}
+  NO_DEFAULT_PATH
+)
+
+if((NOT ONEMKL_INCLUDE_DIR) OR (NOT ONEMKL_LIB_DIR))
+  message(WARNING "oneMKL sdk is incomplete!!")
+  return()
+endif()
+
+set(CMAKE_INCLUDE_PATH ${CMAKE_INCLUDE_PATH}
+    "${ONEMKL_INCLUDE_DIR}")
+set(CMAKE_LIBRARY_PATH ${CMAKE_LIBRARY_PATH}
+    "${ONEMKL_LIB_DIR}")
+
+set(MKL_LIB_NAMES "mkl_intel_lp64" "mkl_gnu_thread" "mkl_core" "mkl_sycl_dft")
+
+foreach(LIB_NAME IN LISTS MKL_LIB_NAMES)
+  find_library(
+    ${LIB_NAME}_library
+    NAMES ${LIB_NAME}
+    HINTS ${ONEMKL_LIB_DIR}
+    NO_CMAKE_PATH
+    NO_CMAKE_ENVIRONMENT_PATH
+  )
+  list(APPEND ONEMKL_XPU_LIBRARIES ${${LIB_NAME}_library})
+endforeach()
+
+list(INSERT ONEMKL_XPU_LIBRARIES 0 "-Wl,--no-as-needed")
+list(APPEND ONEMKL_XPU_LIBRARIES "-Wl,--as-needed")
```

Comment:
`FindONEMKL.cmake` takes responsibility of setting up tool kit.
`MKL.cmake` takes responsibility of implementing the specific building requirements of the current project.
So these two sentences are project specific, please move them two to `MKL.cmake`.

Reply:
Already moved.